elements/lisk-p2p/src/p2p.ts

```diff
@@ -759,26 +764,40 @@ export class P2P extends EventEmitter {
 			random,
 			Math.min(minimumPeerDiscoveryThreshold, knownPeers.length),
 		);
-		/* tslint:enable no-magic-numbers*/
-
-		// TODO: Remove fields that are specific to the current Lisk protocol.
-		const peers = this._pickRandomPeers(randomPeerCount).map(
-			(peerInfo: P2PPeerInfo): ProtocolPeerInfo => {
-				const { ipAddress, ...peerInfoWithoutIp } = peerInfo;
 
-				// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
-				return {
-					...peerInfoWithoutIp,
-					ip: ipAddress,
-				};
-			},
-		);
-		const protocolPeerInfoList: ProtocolPeerInfoList = {
-			success: true,
-			peers,
-		};
-
-		request.end(protocolPeerInfoList);
+		if (request.procedure === REMOTE_RPC_GET_BASIC_PEERS_LIST) {
+			const basicPeers = this._pickRandomPeers(randomPeerCount).map(
+				(peerInfo: P2PPeerInfo): P2PPeerInfo =>
+					// Discovery process only require minmal peers data
+					({
+						ipAddress: peerInfo.ipAddress,
```

Comment:
Shouldn't it return `ip` instead of `ipAddress`?

```suggestion
						ip: peerInfo.ipAddress,
```

Reply:
In this case @lsilvs, we need `ipAddress` instead... @jondubois we should get rid of this inconsistency to avoid misunderstandings.

Reply:
in p2p-lib is ipAddress and in framework context it's `ip`, but yes we need to get rid of this inconsistency as in framework its more prone to errors because of absence of TS ;)